src/__tests__/__snapshots__/fjs_babel-test.js.snap

```diff
@@ -721,6 +721,17 @@ var a: [number, string] = [123, \'duck\'];
 "
 `;
 
+exports[`test type-alias.js 1`] = `
+"type FBID = number;
+type Foo<T> = Bar<T>;
+type Maybe<T> = _Maybe<T, *>;
+
+type union = { type: \'A\' } | { type: \'B\' };
+
+type overloads = (x: string) => number & (x: number) => string;
```

Comment:
The above would resolve to the following in flow:
```
type overloads = (x: string) => (number & (x: number) => string);
```
We need to add the following parenthesis so it looks like:
```
type overloads = ((x: string) => number) & ((x: number) => string);
```

For the general helper i mean (although not sure how it does it for this case): https://github.com/babel/babel/blob/master/packages/babel-generator/src/node/parentheses.js